### PR TITLE
run: delete container's resources once container ends

### LINKED ISSUE #144: runtime standalone: run several containers reduce the boot time progressively
${RUNTIME} --debug --log runtime.log run --pid-file "${BUNDLE}/pid" --bundle "${BUNDLE}" --console $(tty) "${CONTNAME}"

### PR DESCRIPTION
**Pull request by devimc**
fixes #144

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

File: run.go [hunk 1]
```diff
@@ -92,6 +92,12 @@ func run(context *cli.Context) error {
 		if err != nil {
 			return fmt.Errorf("Process state %s: %s", ps.String(), err)
 		}
+
+		// delete container's resources
+		err = delete(pod.GetAllContainers()[0].ID(), true)
```
Comment (sboeuf) [inline run.go]: I would prefer:
```go
if err := delete(container.ID(), true); err != nil {
```
And notice that you could factorize upper in the code something like this:
```go
container := pod.GetAllContainers()[0]
```

Comment (dlespiau) [inline run.go]: That code looks like it will panic if GetAllContainers() returns a empty array for some reason.

Comment (devimc) [inline run.go]: that's not possible, if the container is not created ```create``` will fail returning an error

File: run.go [hunk 2]
```diff
@@ -92,6 +92,11 @@ func run(context *cli.Context) error {
 		if err != nil {
 			return fmt.Errorf("Process state %s: %s", ps.String(), err)
 		}
+
+		// delete container's resources
+		if err := delete(pod.GetAllContainers()[0].ID(), true); err != nil {
```
Comment (sboeuf) [inline run.go]: Thanks for the change, but could you do the factorization too please. And the comment from @dlespiau makes sense, it is always better to check we didn't get `nil` before to use a method from this instance.

Comment (dlespiau) [inline run.go]: If we guarantee that GetAllContainers() will indeed return a non empty array with a non nil element after create, not checking sounds fair enough. I wonder if we do guarantee that (not looked).

**Comment by devimc** [comment]
@sboeuf fixed

**Comment by sboeuf** [comment]
Sounds good :)
LGTM

File: run.go [hunk 3]
```diff
@@ -92,6 +96,11 @@ func run(context *cli.Context) error {
 		if err != nil {
 			return fmt.Errorf("Process state %s: %s", ps.String(), err)
 		}
+
+		// delete container's resources
+		if err := delete(containers[0].ID(), true); err != nil {
```
Comment (jodh-intel) [inline run.go]: Good catch!

File: run.go [hunk 4]
```diff
@@ -82,8 +82,12 @@ func run(context *cli.Context) error {
 
 	detach := context.Bool("detach")
 	if !detach {
-		pid := pod.GetAllContainers()[0].GetPid()
-		p, err := os.FindProcess(pid)
+		containers := pod.GetAllContainers()
+		if len(containers) == 0 {
+			return fmt.Errorf("There are not containers running in the pod: %s", pod.ID())
```
Comment (jodh-intel) [inline run.go]: Nit, but could you change "There are not" to "There are no"?

Comment (devimc) [inline run.go]: fixed

**Comment by jodh-intel** [comment]
lgtm

**Pull request merged**